pkg/resourcemanagers/topology.go

```diff
@@ -32,6 +33,22 @@ import (
 
 const instanceTopologyManagerCacheTimeout = 24 * time.Hour
 
+/*
+We need to ensure that instance types that we expect a response will not successfully complete syncing unless
+we get a response, so we can track known instance types that we expect to get a response for.
+
+Supported instance types for DescribeInstanceTopology as of 2/6/25 from API documentation:
+https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTopology.html
+*/
+var instanceTypePatternsRequireResponse = []*regexp.Regexp{
+	// hpc6a.48xlarge | hpc6id.32xlarge | hpc7a.12xlarge | hpc7a.24xlarge | hpc7a.48xlarge | hpc7a.96xlarge | hpc7g.4xlarge | hpc7g.8xlarge | hpc7g.16xlarge
+	regexp.MustCompile("hpc[0-9]+[a-z]+\\.*"),
+	// p3dn.24xlarge | p4d.24xlarge | p4de.24xlarge | p5.48xlarge | p5e.48xlarge | p5en.48xlarge
+	regexp.MustCompile("p[0-9]+([a-z]+)?\\.*"),
```

Comment:
I wonder if this ends up matching older hosts that don't support the topology API (say anyone still able to launch a p2 / p3) and those hosts can never join a cluster post this change?

Reply:
> those hosts can never join a cluster post this change?

This doesn't require topologies to be returned. It just requires a definitive response, so if there's not topologies available, we'll be happy to move on without setting them.

> I wonder if this ends up matching older hosts that don't support the topology API (say anyone still able to launch a p2 / p3)

Yes, AFAIK, the availability of those instance types is diminishing as new instance types are available, so I didn't think it was worth adding the additional complexity to differentiate. p4 is the oldest instance type listed in pricing: https://aws.amazon.com/ec2/instance-types/